service/src/main/java/uk/nhs/adaptors/gp2gp/transformJsonToXmlTool/XmlSchemaValidator.java

```diff
@@ -0,0 +1,100 @@
+package uk.nhs.adaptors.gp2gp.transformjsontoxmltool;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.io.FilenameUtils;
+import org.springframework.stereotype.Component;
+import org.xml.sax.SAXException;
+import org.xml.sax.SAXParseException;
+import uk.nhs.adaptors.gp2gp.common.configuration.RedactionsContext;
+
+import javax.xml.transform.stream.StreamSource;
+import javax.xml.validation.SchemaFactory;
+import javax.xml.validation.Validator;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.StringReader;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Paths;
+
+@Component
+@Slf4j
+@RequiredArgsConstructor()
+public class XmlSchemaValidator {
+    private final RedactionsContext redactionsContext;
+
+    private static final String SCHEMA_PATH = "../service/src/test/resources/mim/Schemas/";
+    private static final String RCMR_IN030000UK06_SCHEMA_PATH =  SCHEMA_PATH + RedactionsContext.NON_REDACTION_INTERACTION_ID + ".xsd";
+    private static final String RCMR_IN030000UK07_SCHEMA_PATH =  SCHEMA_PATH + RedactionsContext.REDACTION_INTERACTION_ID + ".xsd";
+    private static final String OUTPUT_PATH =
+        Paths.get("src/").toFile().getAbsoluteFile().getAbsolutePath() + "/../../transformJsonToXml/output/";
+
+    public void validateOutputToXmlSchema(String filename, String xmlResult) {
+        LOGGER.info("Validating {} against {} schema", filename, RedactionsContext.REDACTION_INTERACTION_ID);
+
+        var xsdErrorHandler = new uk.nhs.adaptors.gp2gp.transformjsontoxmltool.XsdErrorHandler();
+        Validator xmlValidator;
+
+        try {
+            xmlValidator = getXmlValidator(xsdErrorHandler);
+        } catch (SAXException e) {
+            LOGGER.info("Could not load schema file for {} context.", RedactionsContext.REDACTION_INTERACTION_ID);
```

Comment:
as it is an exception caught here it should be wrapped into "LOGGER.error" (and not info)